apps/api/src/job-posting/repository.js

```diff
@@ -70,64 +57,61 @@ const getFilteredJobPostingsQuery = (client, filters, sort) => {
             'organization.id': 'job_posting.hiring_organization_id',
         });
 
-    if (title) {
-        query.andWhere('title', 'LIKE', `%${title}%`);
-    }
-    if (skills) {
-        query.andWhere('skills', 'LIKE', `%${skills}%`);
-    }
-    if (employmentType) {
-        query.andWhere('employmentType', employmentType);
-    }
-    if (hiringOrganizationPostalCode) {
-        query.andWhere(
-            'organization.postal_code',
-            'LIKE',
-            `${hiringOrganizationPostalCode}%`
-        );
-    }
-    if (hiringOrganizationName) {
-        query.andWhere(
-            'organization.name',
-            'LIKE',
-            `%${hiringOrganizationName}%`
-        );
-    }
-    if (hiringOrganizationAddressLocality) {
-        query.andWhere(
-            'organization.address_locality',
-            'LIKE',
-            `${hiringOrganizationAddressLocality}%`
-        );
-    }
-    if (hiringOrganizationAddressCountry) {
-        query.andWhere(
-            'organization.address_country',
-            'LIKE',
-            `${hiringOrganizationAddressCountry}%`
-        );
-    }
+    filters.map((filter) => {
+        switch (filter.name) {
```

Comment:
N'étant pas fan des switchs, je privilégie l'approche fonctionnelle avec un objet : https://medium.com/chrisburgin/rewriting-javascript-replacing-the-switch-statement-cfff707cf045

Reply:
Je connaissais pas je vais me pencher dessus !

Reply:
C'est pas obligatoire hein :stuck_out_tongue: mais il y a quelques arguments et surtout une approche plus fonctionnelle !

Reply:
Okay j'ai réussi à faire ça :

Reply:
Ça me parait bien !